src/api/selection.js

```diff
@@ -1,202 +1,118 @@
+        nodeHelper.removeNode(marker.previousSibling);
+      }
+    }
 
+    /**
+     * Wrapper for object holding currently selected text.
+     */
+    function Selection() {
       this.selection = rootDoc.getSelection();
       if (this.selection.rangeCount && this.selection.anchorNode) {
         // create the range to avoid chrome bug from getRangeAt / window.getSelection()
         // https://code.google.com/p/chromium/issues/detail?id=380690
         this.range = document.createRange();
-        var reverseRange = document.createRange();
-
-        this.range.setStart(this.selection.anchorNode, this.selection.anchorOffset);
-        reverseRange.setStart(this.selection.focusNode, this.selection.focusOffset);
 
-        // Check if anchorNode is before focusNode, use reverseRange if not
-        if (this.range.compareBoundaryPoints(Range.START_TO_START, reverseRange) <= 0) {
+        if( nodeHelper.isBefore(this.selection.anchorNode, this.selection.focusNode)
+          || this.selection.anchorNode === this.selection.focusNode ) {
+          this.range.setStart(this.selection.anchorNode, this.selection.anchorOffset);
           this.range.setEnd(this.selection.focusNode, this.selection.focusOffset);
-        }
-        else {
-          this.range = reverseRange;
+        } else {
+          this.range.setStart(this.selection.focusNode, this.selection.focusOffset);
           this.range.setEnd(this.selection.anchorNode, this.selection.anchorOffset);
         }
       }
     }
 
     /**
-     * @returns Closest ancestor Node satisfying nodeFilter. Undefined if none exist before reaching Scribe container.
+     * @returns Closest ancestor Node satisfying nodeFilter. Undefined if none
+     * exist before reaching Scribe container.
      */
     Selection.prototype.getContaining = function (nodeFilter) {
       var range = this.range;
       if (!range) { return; }
 
       var node = new scribe.api.Node(this.range.commonAncestorContainer);
-      var isTopContainerElement = node.node && scribe.el === node.node;
-
-      return ! isTopContainerElement && nodeFilter(node.node) ? node.node : node.getAncestor(scribe.el, nodeFilter);
+      return ! (node.node && scribe.el === node.node) && nodeFilter(node.node) ?
+        node.node :
+        node.getAncestor(scribe.el, nodeFilter);
     };
 
     Selection.prototype.placeMarkers = function () {
       var range = this.range;
-      if (!range) {
-        return;
-      }
+      if (!range) { return; }
 
-      //we need to ensure that the scribe's element lives within the current document to avoid errors with the range comparison (see below)
-      //one way to do this is to check if it's visible (is this the best way?).
-      if (!scribe.el.offsetParent) {
+      //we need to ensure that the scribe's element lives within the current document
+      //to avoid errors with the range comparison (see below)
+      if (!document.contains(scribe.el)) {
```

Comment:
@rrees I think yes, as per the comment just above: if the element is not inside the document, then quit.